pkg/apis/performanceprofile/v2/performanceprofile_validation.go

```diff
@@ -201,31 +238,175 @@ func (r *PerformanceProfile) validateSelectors() field.ErrorList {
+				r.Spec.NodeSelector,
+				fmt.Sprintf("Failed to detect architecture for node %s", nodes.Items[0].Status.NodeInfo.MachineID),
+			),
+		)
+
+		// If we failed to detect cpu architefcture there is not much point to continue
+		// We would likely just get an error for every single node with the same error
+		return allErrs
+	}
+
+	// Make sure all other nodes have the same value
+	for i := 1; i < len(nodes.Items); i++ {
+		actualArchitecture := getCpuArchitectureForNode(nodes.Items[i])
+		if actualArchitecture != expectedArchitecture {
+			allErrs = append(allErrs,
+				field.Invalid(
+					field.NewPath("spec.nodeSelector"),
+					r.Spec.NodeSelector,
+					fmt.Sprintf("Node %s has architecture %s but was expecting %s", nodes.Items[i].Status.NodeInfo.MachineID, actualArchitecture, expectedArchitecture),
+				),
+			)
+		}
+	}
+
+	return allErrs
+}
+
+func getCpuArchitectureForNode(node corev1.Node) string {
+	return node.Status.NodeInfo.Architecture
+}
+
+func (r *PerformanceProfile) validateAllNodesAreSameCpuCapacity(nodes corev1.NodeList) field.ErrorList {
+	var allErrs field.ErrorList
+
+	// We need to use one of the nodes as a reference for comparing against the rest
+	// The first item in the list is simple and easy to use
+	expectedCpuCapacity := getCpuCapacityForNode(nodes.Items[0])
+
+	if expectedCpuCapacity == "" {
+		allErrs = append(allErrs, 
+			field.Invalid(
+				field.NewPath("spec.nodeSelector"),
+				r.Spec.NodeSelector,
+				fmt.Sprintf("Failed to detect cpu capacity for node %s", nodes.Items[0].Status.NodeInfo.MachineID),
+			),
+		)
+
+		// If we failed to detect cpu capacity there is not much point to continue
+		// We would likely just get an error for every single node with the same error
+		return allErrs
+	}
+
+	// Make sure all other nodes have the same value
+	for i := 1; i < len(nodes.Items); i++ {
+		actualCpuCapacity := getCpuCapacityForNode(nodes.Items[i])
+		if actualCpuCapacity != expectedCpuCapacity {
+			allErrs = append(allErrs,
+				field.Invalid(
+					field.NewPath("spec.nodeSelector"),
+					r.Spec.NodeSelector,
+					fmt.Sprintf("Node %s has CPU capacity %s but was expecting %s", nodes.Items[i].Status.NodeInfo.MachineID, actualCpuCapacity, expectedCpuCapacity),
+				),
+			)
+		}
+	}
+
+	return allErrs
+}
+
+func getCpuCapacityForNode(node corev1.Node) string {
+	return node.Status.Capacity.Cpu().String()
+}
+
+func (r *PerformanceProfile) validateHugePages(nodes corev1.NodeList) field.ErrorList {
 	var allErrs field.ErrorList
 
 	if r.Spec.HugePages == nil {
 		return allErrs
 	}
 
-	// validate that default hugepages size has correct value, currently we support only 2M and 1G(x86_64 architecture)
+    // `validateHugePages` implicitly relies on `validateAllNodesAreSameCpuArchitecture` to have already been run
+    // Under that assumption we can return any node from the list since they should all be the same architecture
+    // However it is simple and easy to just return the first node
```

Comment:
Indentation looks off - somehow spaces crept in.

Reply:
Fixed in latest commit